cli/tests/pcluster/test_utils.py

```diff
@@ -607,3 +607,114 @@ async def async_method(self, param):
 def test_is_feature_supported(feature, region, expected_result):
     actual_result = utils.is_feature_supported(feature=feature, region=region)
     assert_that(actual_result).is_equal_to(expected_result)
+
+
+@pytest.mark.parametrize(
+    "service_name, partition, region, regional, expected_result, expected_exception",
+    [
+        pytest.param(
+            "logs",
+            "aws",
+            None,
+            False,
+            "logs.amazonaws.com",
+            None,
+            id="Non-regional service principal in aws partition",
+        ),
+        pytest.param(
+            "logs",
+            "aws",
+            "us-east-1",
+            True,
+            "logs.us-east-1.amazonaws.com",
+            None,
+            id="Regional service principal in aws partition",
+        ),
+        pytest.param(
+            "logs",
+            "aws",
+            None,
+            True,
+            None,
+            ValueError("Region must be provided when 'regional' is True."),
+            id="Regional=True without region provided",
+        ),
+        pytest.param(
+            "logs",
+            "aws-cn",
+            None,
+            False,
+            "logs.amazonaws.com.cn",
+            None,
+            id="Non-regional service principal in aws-cn partition",
+        ),
+        pytest.param(
+            "logs",
+            "aws-cn",
+            "cn-north-1",
+            True,
+            "logs.cn-north-1.amazonaws.com.cn",
+            None,
+            id="Regional service principal in aws-cn partition",
+        ),
+    ],
+)
+def test_get_service_principal(mocker, service_name, partition, region, regional, expected_result, expected_exception):
+    """Test get_service_principal with various inputs."""
+    # Mock get_url_domain_suffix
+    domain_suffix = "amazonaws.com.cn" if partition == "aws-cn" else "amazonaws.com"
```

Comment:
*Test Best Practice*
This if..else logic is not necessary.
Here you're unit testing the function `get_service_principal`, which does not make any assumption on the value returned by `get_url_domain_suffix`. So the mock should return a static value `AWS_DOMAIN_SUFFIX` and the test should assert that the function `get_url_domain_suffix`is called once

Reply:
You are right, I should not mock `get_url_domain_suffix`. I re-checked the code and I decided to remove the mock logic and let the function itself handle the `domain_suffix`. Does it make sense to you?